controllers/configmap_controller.go

```diff
@@ -168,6 +168,29 @@ func validateAddress(address string) error {
 	return nil
 }
 
+// getNodeUsername retrieves the username associated with the given node from the instance ConfigMap data
+func getNodeUsername(configMapData map[string]string, node *core.Node) (string, error) {
+	if node == nil {
+		return "", errors.New("cannot get username for nil node")
+	}
+	// Find entry in ConfigMap that is associated to node via address
+	for _, address := range node.Status.Addresses {
+		if data, found := configMapData[address.Address]; found {
+			return extractUsernameFromEntry(data)
+		}
+	}
+	return "", errors.Errorf("unable to find instance associated with node %s", node.GetName())
+}
+
+// extractUsernameFromEntry returns the username string from data in the form username=<username>
+func extractUsernameFromEntry(value string) (string, error) {
```

Comment:
s/value/data

Reply:
Would be nice to unit test `func extractUsernameFromEntry(value string) (string, error) {}`.

Reply:
`value` refers to the second part of the key/value pair that makes up each ConfigMap entry. From our README: 
> Each entry in the data section of the ConfigMap should be formatted with the address as the key, and a value with the format of username=<username>

IMO `extractUsernameFromEntry` does not need separate tests as there are already unit tests for `getNodeUsername`. It is just a helper for `getNodeUsername` and moreover is not used in unit tests (or elsewhere).

I will leave this is as.